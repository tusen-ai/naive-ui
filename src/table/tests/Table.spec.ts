import { mount } from '@vue/test-utils'
import { NTable, NTbody, NThead, NTd, NTh, NTr } from '../index'

describe('n-table', () => {
  it('should work with import on demand', () => {
    mount(NTable)
  })

  it('should work with `bottom-bordered` prop', async () => {
    const wrapper = mount(NTable)
    expect(wrapper.find('.n-table').classes()).toContain(
      'n-table--bottom-bordered'
    )

    await wrapper.setProps({ bottomBordered: false })
    expect(wrapper.find('.n-table').classes()).not.toContain(
      'n-table--bottom-bordered'
    )
  })

  it('should work with `bordered` prop', async () => {
    const wrapper = mount(NTable)
    expect(wrapper.find('.n-table').classes()).toContain('n-table--bordered')

    await wrapper.setProps({ bordered: false })
    expect(wrapper.find('.n-table').classes()).not.toContain(
      'n-table--bordered'
    )
  })

  it('should work with `single-column` prop', async () => {
    const wrapper = mount(NTable)
    expect(wrapper.find('.n-table').classes()).not.toContain(
      'n-table--single-column'
    )

    await wrapper.setProps({ singleColumn: true })
    expect(wrapper.find('.n-table').classes()).toContain(
      'n-table--single-column'
    )
  })

  it('should work with `single-line` prop', async () => {
    const wrapper = mount(NTable)
    expect(wrapper.find('.n-table').classes()).toContain('n-table--single-line')

    await wrapper.setProps({ singleLine: false })
    expect(wrapper.find('.n-table').classes()).not.toContain(
      'n-table--single-line'
    )
  })

  it('should work with `size` prop', async () => {
    ;(['small', 'medium', 'large'] as const).forEach((size) => {
      const wrapper = mount(NTable, { props: { size } })
      expect(wrapper.find('.n-table').attributes('style')).toMatchSnapshot()
    })
  })

  it('should work with `striped` prop', async () => {
    const wrapper = mount(NTable)
    expect(wrapper.find('.n-table').classes()).not.toContain('n-table--striped')

    await wrapper.setProps({ striped: true })
    expect(wrapper.find('.n-table').classes()).toContain('n-table--striped')
  })
})

describe('n-table-body', () => {
  it('should work with import on demand', () => {
    mount(NTbody)
  })
})

describe('n-table-head', () => {
  it('should work with import on demand', () => {
    mount(NThead)
  })
})

describe('n-table-td', () => {
  it('should work with import on demand', () => {
    mount(NTd)
  })
})

describe('n-table-th', () => {
  it('should work with import on demand', () => {
    mount(NTh)
  })
})

describe('n-table-tr', () => {
  it('should work with import on demand', () => {
    mount(NTr)
  })
})
