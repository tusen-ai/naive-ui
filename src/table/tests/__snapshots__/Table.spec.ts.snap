// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`n-table > should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-td-color: #fff; --n-td-color-modal: #fff; --n-td-color-popover: #fff; --n-td-text-color: rgb(51, 54, 57); --n-border-color: rgba(239, 239, 245, 1); --n-border-color-modal: rgba(239, 239, 245, 1); --n-border-color-popover: rgba(239, 239, 245, 1); --n-border-radius: 3px; --n-font-size: 14px; --n-th-color: rgba(250, 250, 252, 1); --n-th-color-modal: rgba(250, 250, 252, 1); --n-th-color-popover: rgba(250, 250, 252, 1); --n-th-font-weight: 500; --n-th-text-color: rgb(31, 34, 37); --n-line-height: 1.6; --n-td-padding: 6px; --n-th-padding: 6px; --n-td-color-striped: rgba(250, 250, 252, 1); --n-td-color-striped-modal: rgba(250, 250, 252, 1); --n-td-color-striped-popover: rgba(250, 250, 252, 1);"`;

exports[`n-table > should work with \`size\` prop 2`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-td-color: #fff; --n-td-color-modal: #fff; --n-td-color-popover: #fff; --n-td-text-color: rgb(51, 54, 57); --n-border-color: rgba(239, 239, 245, 1); --n-border-color-modal: rgba(239, 239, 245, 1); --n-border-color-popover: rgba(239, 239, 245, 1); --n-border-radius: 3px; --n-font-size: 14px; --n-th-color: rgba(250, 250, 252, 1); --n-th-color-modal: rgba(250, 250, 252, 1); --n-th-color-popover: rgba(250, 250, 252, 1); --n-th-font-weight: 500; --n-th-text-color: rgb(31, 34, 37); --n-line-height: 1.6; --n-td-padding: 12px; --n-th-padding: 12px; --n-td-color-striped: rgba(250, 250, 252, 1); --n-td-color-striped-modal: rgba(250, 250, 252, 1); --n-td-color-striped-popover: rgba(250, 250, 252, 1);"`;

exports[`n-table > should work with \`size\` prop 3`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-td-color: #fff; --n-td-color-modal: #fff; --n-td-color-popover: #fff; --n-td-text-color: rgb(51, 54, 57); --n-border-color: rgba(239, 239, 245, 1); --n-border-color-modal: rgba(239, 239, 245, 1); --n-border-color-popover: rgba(239, 239, 245, 1); --n-border-radius: 3px; --n-font-size: 15px; --n-th-color: rgba(250, 250, 252, 1); --n-th-color-modal: rgba(250, 250, 252, 1); --n-th-color-popover: rgba(250, 250, 252, 1); --n-th-font-weight: 500; --n-th-text-color: rgb(31, 34, 37); --n-line-height: 1.6; --n-td-padding: 12px; --n-th-padding: 12px; --n-td-color-striped: rgba(250, 250, 252, 1); --n-td-color-striped-modal: rgba(250, 250, 252, 1); --n-td-color-striped-popover: rgba(250, 250, 252, 1);"`;
